aws/defaults/handlers.go

```diff
@@ -252,3 +269,31 @@ var ValidateEndpointHandler = aws.NamedHandler{Name: "core.ValidateEndpointHandl
 		r.Error = aws.ErrMissingEndpoint
 	}
 }}
+
+// AttemptClockSkewHandler records the estimated clock skew between the client
+// and service response clocks. This estimation will be no more granular than
+// one second. It will not be populated until after at least the first
+// attempt's response is received.
+var AttemptClockSkewHandler = aws.NamedHandler{
+	Name: "core.AttemptClockSkewHandler",
+	Fn: func(r *aws.Request) {
+		if r.ResponseAt.IsZero() || r.HTTPResponse == nil || r.HTTPResponse.StatusCode == 0 {
+			return
+		}
+
+		respDateHeader := r.HTTPResponse.Header.Get("Date")
+		if len(respDateHeader) == 0 {
+			return
+		}
+
+		respDate, err := http.ParseTime(respDateHeader)
+		if err != nil {
+			// unable to parse response error for some reason.
```

Comment:
err can be logged, if we don't want to put it on request.

Reply:
can add logging here